static/search.js

```diff
@@ -674,13 +674,45 @@ var Search = Control.extend({
 	// function activateResult
 	// sets property and adds class to active result
 	activateResult: function($result){
+		// Get position top of last active element
+		var lastResultPosTop = parseInt(this.$activeResult && this.$activeResult.position().top || 0);
 		this.deactivateResult();
 		this.$activeResult = $result;
 		this.$activeResult.addClass(this.options.keyboardActiveClass);
 
-		var activeResultOffset = this.getActiveResultOffset();
+		// Get position top of current active element
+		var activeResultPosTop = parseInt(this.$activeResult.position().top);
+		var resultsContainerHeight = this.$resultsContainer.height();
 
-		this.$resultsContainer.scrollTop(this.$activeResult.position().top - activeResultOffset);
+		var isMovingDown = lastResultPosTop < this.$activeResult.position().top;
```

Comment:
Add a comment here to clarify that you are talking about the arrow key direction